src/blobifier.py

```diff
@@ -48,28 +63,33 @@ def _fill_blobs(self, mask: np.ndarray) -> np.ndarray:
 
         def choose_value(x):
             center = x[4]
-            selection = x[x != -1]
-            if center == -1 and not selection.size == 0:
+            selection = x[x != NULL_VALUE]
+            if center == NULL_VALUE and not selection.size == 0:
                 m = mode(selection, axis=None).mode
                 return m
             else:
                 return center
 
         blob_raster = self.data.copy()
-        blob_raster[mask] = -1
-        while np.any(blob_raster == -1):
+        blob_raster[mask] = NULL_VALUE
+        while np.any(blob_raster == NULL_VALUE):
             result = ndimage.generic_filter(
                 blob_raster,
                 choose_value,
                 footprint=neighborhood,
                 mode="constant",
-                cval=0
+                cval=NULL_VALUE,
```

Comment:
This fixes a bug. If a pixel is on the border, it won't have all 8 neighbors. Instead, it gets the values of the neighbors it does have and the `cval` for each neighbor that does not exist. If the `cval` is 0, that could be chosen as the final value of the pixel if it is the mode. However, if it is the `NULL_VALUE`, then it will never be chosen.